skbuild/setuptools_wrap.py

```diff
@@ -440,9 +440,15 @@ def setup(*args, **kw):  # noqa: C901
     try:
         cmkr = cmaker.CMaker()
         if not skip_cmake:
-            env = cmkr.configure(cmake_args,
-                                 cmake_source_dir=cmake_source_dir,
-                                 cmake_install_dir=skbuild_kw['cmake_install_dir'])
+            # skip the configure step for a cached build
+            env = cmkr.get_cached_env()
+            if env is None:
+                env = cmkr.configure(
+                    cmake_args,
```

Comment:
One last thing would be to serialize the `cmake_args` into a file and compare it didn't change.

May a function  `get_cached_cmake_args` could be added, then configuration would be done only if the ``env is None` or `cmake_args != cached_cmake_args`